### DISCUSSION TITLE
How to protect your public GitHub repository from unauthorized overwriting?

### DISCUSSION THREAD

**Discussion by MateuszPodeszwa** [community/community#53358] [New to GitHub] [answered]
Hello, as a newcomer, I would like to ask for advice on how to prevent unauthorized overwriting of my public repository. Specifically, I want to avoid situations where someone could clone my repository using CLI, make changes to it, and push those changes to my official GitHub repository without my approval.

**Comment by xy3** [comment, marked as answer]
Nobody can push changes to your branch without forking your repository, opening a pull request and getting your approval, that's by default.

If you want to add more protection rules for specific branches, you can click on the Settings tab on your repository, go to the Branches section and click "Add branch protection rule".

![image](https://user-images.githubusercontent.com/32339757/233586387-45e446a6-6d7d-4abc-8914-08aa9024ea73.png)

You can read more about protected branches here: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches
